geomstats/visualization/CategoricalDistributionsManifold.py

```diff
@@ -0,0 +1,484 @@
+            z = [0, 0, 0, 1]
+            vertices = [[0, 1, 2], [0, 1, 3], [0, 2, 3], [1, 2, 3]]
+            tupleList = list(zip(x, y, z))
+            poly3d = [
+                [tupleList[vertices[ix][iy]] for iy in range(len(vertices[0]))]
+                for ix in range(len(vertices))
+            ]
+            self.ax.add_collection3d(
+                Poly3DCollection(
+                    poly3d,
+                    edgecolors="k",
+                    facecolors=(0.9, 0.9, 0.9, 1.0),
+                    linewidths=3,
+                    alpha=0.2,
+                )
+            )
+
+        elif self.dim == 2:
+            X = np.linspace(start=min_limit, stop=max_limit, num=101,
+                            endpoint=True)
+            Y = 1 - X
+            self.ax.fill_between(X, Y, color=(0.9, 0.9, 0.9, 1.0))
+            self.ax.set_title("2 Dimension Categorical Manifold")
+
+    def set_points(self, points):
+        self.points = points
+
+    def scatter(self, n_samples, **scatter_kwargs):
+        """Scatter plot some randomly sampled points in the manifold.
+
+        Plot the manifold along with some randomly sampled points
+        lying on the manifold.
+
+        Parameters:
+        -----------
+        n_samples : integer
+            The number of randomly sampled points.
+
+        **scatter_kwargs: optional
+            Inherits the matplotlib scatter function parameters.
+
+        Returns:
+        --------
+        None.
+
+        Notes:
+        ------
+        This method internally calls the plot method.
+        """
+        self.set_points(self.dist.random_point(n_samples=n_samples))
+        self.plot()
+        if self.dim == 2:
+
+            for point in self.points:
+                self.ax.scatter(point[0], point[1], **scatter_kwargs)
+            self.ax.set_title(
+                f"2 Dimension Categorical Manifold with {n_samples} Samples"
+            )
+        elif self.dim == 3:
+            for point in self.points:
+                self.ax.scatter(point[0], point[1], point[2], **scatter_kwargs)
+            self.ax.set_title(
+                f"3 Dimension Categorical Manifold with {n_samples} Samples"
+            )
+
+        self.clear_points()
+
+    def plot_geodesic(self, initial_point, end_point=None,
+                      tangent_vector=None):
+        """Plot a geodesic on the manifold.
+
+        Plot a geodesic that is either specified with
+        1) an initial_point and an end_point, or
+        2) an initial point and an initial tangent vector
+        on the manifold.
+
+        Parameters:
+        -----------
+        initial_point: array-like, shape = [..., dim + 1]
+            Initial point on the manifold.
+
+        end_point: optional, array-like, shape = [..., dim + 1]
+            End point on the manifold.
+
+        tangent_vector: optional, array-like, shape = [..., dim + 1]
+            Initial tangent vector at the initial point.
+
+        Returns:
```

Comment:
Remove : and corresponding underline

Reply:
(Do it everything in the files of this PR, I will stop copy-pasting that comment everwyhere)